build.gradle

```diff
@@ -1,78 +1,265 @@
+    from sourceSets.test.output
+}
+
+tasks.withType(Test) {
+    useJUnitPlatform()
+
+    testLogging {
+        exceptionFormat = 'full'
     }
 }
 
-jar {
-    baseName = "civil-sdt-common"
+artifacts {
+    archives sourcesJar
+    archives javadocJar
+    archives testJar
 }
 
+bootJar {
+    enabled = false
+}
 
 dependencies {
-    // Use JUnit Jupiter API for testing.
-    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.2'
 
-    // Use JUnit Jupiter Engine for testing.
-    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine'
+    // This dependency is used internally, and not exposed to consumers on their own compile classpath.
+    implementation 'com.google.guava:guava:30.0-jre'
+    implementation group: 'org.apache.logging.log4j', name: 'log4j-api', version: log4jVersion
+    implementation group: 'org.apache.logging.log4j', name: 'log4j-core', version: log4jVersion
+    implementation 'org.apache.logging.log4j:log4j-slf4j2-impl:2.20.0'
+    implementation group: 'org.springdoc', name: 'springdoc-openapi-ui', version: '1.6.7'
+    implementation group: 'org.springframework', name: 'spring-jms', version: '5.3.23'
+    implementation group: 'org.apache.cxf', name: 'cxf-spring-boot-starter-jaxws', version: cxfVersion
+    implementation group: 'org.apache.cxf', name: 'cxf-rt-transports-http-jetty', version: cxfVersion
+    implementation group: 'org.apache.cxf', name: 'cxf-rt-features-logging', version: cxfVersion
+    implementation group: 'javax.jws', name: 'javax.jws-api', version: '1.1'
+    implementation group: 'org.springframework.boot', name: 'spring-boot-starter-web-services'
+    implementation group: 'org.springframework.boot', name: 'spring-boot-starter-activemq'
+    implementation 'org.hibernate:hibernate-core:5.6.15.Final'
+    implementation group: 'xerces', name: 'xercesImpl', version: '2.12.2'
+    implementation(group: 'org.springframework.boot', name: 'spring-boot-starter-jdbc') {
+        exclude group: 'org.apache.tomcat', module: 'tomcat-jdbc'
+    }
+
+    testImplementation libraries.junit5
+    testImplementation group: 'info.solidsoft.gradle.pitest', name: 'gradle-pitest-plugin', version: '1.4.0'
+    testImplementation group: 'org.easymock', name: 'easymock', version: '5.0.1'
+    testImplementation group: 'org.jdom', name: 'jdom2', version: '2.0.6.1'
+    testImplementation 'org.mockito:mockito-core:5.2.0'
+    testImplementation group: 'org.mockito', name: 'mockito-inline', version: '3.11.1'
+    testImplementation group: 'org.powermock', name: 'powermock-api-mockito2', version: '2.0.9'
+    testImplementation group: 'org.powermock', name: 'powermock-module-junit4', version: '2.0.9'
+    testImplementation group: 'org.dbunit', name: 'dbunit', version: '2.7.3'
+    testImplementation 'org.springframework:spring-test:6.0.7'
+    testImplementation group: 'xerces', name: 'xercesImpl', version: '2.12.2'
+    testImplementation 'org.apache.logging.log4j:log4j-slf4j2-impl:2.20.0'
```

Comment:
Should this use the log4jVersion property for the version number instead of the hard coded value?

Reply:
done

Reply:
Confirmed